### ISSUE TITLE
Relative path in import map appears to break package.json dependency resolution

### ISSUE THREAD

**Issue by michaelhthomas** [denoland/deno#26624]
Version: Deno 2.0.2

I'm aiming to use a relative path in an import map, something like the following:

```json
"imports": {
    "$self": "./src/index.ts"
}
```

However, adding the line above causes dependencies from `package.json` to no longer be resolved properly when running `deno install`. I've attached a minimal reproduction below:
https://codesandbox.io/p/devbox/rpz8h9

Adding the import map above leads to an error like the following:
```
error: Relative import path "is-odd" not prefixed with / or ./ or ../ and not in import map from "file:///project/workspace/index.ts"
    at file:///project/workspace/index.ts:1:19
```

where there previously would have been no errors (since `is-odd` is specified in `package.json`).

**Comment by nathanwhit** [comment]
Looks like this was actually fixed in a refactor as part of #26558.

You can update to the latest canary version (`deno upgrade canary`) or wait until the next patch release (should be next release) to get the fix.

**Comment by michaelhthomas** [comment]
That solved my issue, thanks for your help!
